nsqlookupd/http.go

```diff
@@ -223,6 +224,30 @@ func (s *httpServer) doClusterStats(w http.ResponseWriter, req *http.Request, ps
 
 func (s *httpServer) doTopics(w http.ResponseWriter, req *http.Request, ps httprouter.Params) (interface{}, error) {
 	topics := s.ctx.nsqlookupd.DB.FindTopics()
+	//wrap topic meta info
+	reqParams, err := url.ParseQuery(req.URL.RawQuery)
+	if err != nil {
+		return nil, http_api.Err{400, "INVALID_REQUEST"}
+	}
+	if reqParams.Get("metaInfo") == "true" && s.ctx.nsqlookupd.coordinator != nil {
+		var topicsInfo []*clusterinfo.TopicInfo
+		for _, topic := range topics {
```

Comment:
I think it will be slow here to get all the topic meta. We need cache the results and invalid the cache carefully while topic meta updated.